test/test_averaged_field.jl

```diff
@@ -23,19 +23,19 @@ using Oceananigans.Grids: halo_size
                 set!(T, trilinear)
                 set!(w, trilinear)
 
-                @compute T̃ = mean(T, dims=(1, 2, 3))
```

Comment:
In `abstract_field.jl` mean is defined and if I understand will still be defined after merger.  I'm a bit confused as to why we are removing `mean` but addign in `AveagedField` in this script here.  Sorry if I'm missing the obvious.

Reply:
Ah we used to abuse `mean` a little bit.

It used be an alias for `AveragedField` so `mean(::Field)` would actually return an `AveragedField`, which is cool but maybe not what you expect after using the vanilla `Statistics.mean`.

Now `mean` returns a number so I had to change `mean` to `AveragedField` here.